src/Core/Resolvers/CosmosMutationEngine.cs

```diff
@@ -258,6 +273,29 @@ private static async Task<ItemResponse<JObject>> HandleUpdateAsync(IDictionary<s
             return item;
         }
 
+        /// <summary>
+        /// Helper method to get the role with which the GraphQL API request was executed.
+        /// </summary>
+        /// <param name="context">HotChocolate context for the GraphQL request</param>
+        private static string GetRoleOfGraphQLRequest(IMiddlewareContext context)
+        {
+            string role = string.Empty;
+            if (context.ContextData.TryGetValue(key: AuthorizationResolver.CLIENT_ROLE_HEADER, out object? value) && value is StringValues stringVals)
```

Comment:
`AuthorizeMutationFields()` method already contains logic to extract the role. For SQL, I refactored the logic for extracting the role from graphQL context to a separate helper function. It could be worth checking if that can be re-used here.

Reply:
Sure. I'll check that

Reply:
I examined the AuthorizeMutationFields, and it indeed retrieves the role. While I could store that information in a global variable and access it within my scenario, I believe that approach might not be clean and may not effectively communicate the intention behind the change. I am more inclined to obtain the role from this helper function. Thoughts?

Reply:
`AuthorizeMutationFields()` should be updated to now call `GetRoleOfGraphQLRequest()` to eliminate code duplication.